### ISSUE TITLE
Named servers broken in 4.1.0

### ISSUE THREAD

**Issue by benz0li** [jupyterhub/jupyterhub#4749]
### Bug description

Spawning multiple servers (containers) results in 'too many redirects' error.

### How to reproduce

A JupyterHub + multiple JupyterLab images with JupyterHub 4.1.1 installed.

#### Expected behaviour

No issue running multiple servers (containers) in parallel.

#### Actual behaviour

Logs of a JupyterLab container: Repeated

```
[I 2024-03-24 13:57:09.790 ServerApp] 302 GET /user/benz0li/qgis-latest/lab -> /user/benz0li/qgis-latest/lab (benz0li@[redacted]) 6.93ms
[W 2024-03-24 13:57:09.821 ServerApp] Attempting to clear invalid _xsrf cookie 'Mnwx...'
```

👉 JupyterLab (`/lab`) not loading.

### Your personal set up

JupyterHub: [`glcr.b-data.ch/jupyterhub/jupyterhub:4.1.1-oauth-docker`](https://gitlab.b-data.ch/jupyterhub/jupyterhub/container_registry)

JupyterLab:

* [JupyterLab QGIS docker stack](https://github.com/b-data/jupyterlab-qgis-docker-stack)
* [JupyterLab Python docker stack](https://github.com/b-data/jupyterlab-python-docker-stack)
* [JupyterLab Julia docker stack](https://github.com/b-data/jupyterlab-julia-docker-stack)
* [JupyterLab R docker stack](https://github.com/b-data/jupyterlab-r-docker-stack)

ℹ️ base-images built with `JUPYTERHUB_VERSION=4.1.1`.

Ping @minrk

**Comment by minrk** [comment]
Thanks for reporting! I think I know what this is, I will try to fix it.

**Comment by minrk** [comment]
closed by #4750, will make 4.1.2 shortly.

**Comment by benz0li** [comment]
I will wait for the next 4.1 release(s) and further feedback from the community.

Cross reference: https://github.com/jupyterhub/jupyterhub/issues/4752

**Comment by minrk** [comment]
4.1.3 is published fixing #4752. Thanks for testing! Let us know if you hit any further issues.

**Comment by benz0li** [comment]
> Let us know if you hit any further issues.

Looks good so far.
